![image](https://github.com/Naouri17/profile/assets/40064898/bf74912c-56d5-40c4-932f-6980f371bd20)# Researcher & Mobile Developer

#### Summary: 
Recent graduate with a PhD degree in Computer Science from the University of Science and Technology of Beijing, 
bringing a blend of fieldwork expertise and data-analysis proficiency in IoT, cloud computing, fog computing, 
edge computing, federated learning, Social Internet of Things, and more. My passion has recently evolved towards AI application, 
where I aspire to contribute to the development of innovative modeling techniques for computing ecosystems.

#### Publication Statistics:
Cumulative Impact Factor: **60.5**
Total Citations: **206** <a href="https://scholar.google.com/citations?user=19TeXs8AAAAJ&hl=en&oi=ao" target="_blank">Google Scholar</a>

#### Technical Skills:

- Programming Languages: _Proficient in R, Python, Linux command line; experience with Shiny, Google Colab._
- Phenomics and Genomics: _Expertise in molecular breeding techniques, NGS data analysis, QTL/GWAS, genomic selection, designing field trials, managing phenotyping pipelines, and analyzing high-throughput phenotypic and genotypic data._
- Data Science: _Multivariate analysis, machine learning, large scale genomic data analysis and visualization._
  
## üéì Education
- **Ph.D.**, Optimizing Resource Allocation in Fog and Edge Computing Environments | <a href="https://tnau.ac.in/" target="_blank">University of Science and Technology Beijing. </a> (_SEP 2018 - Thesis submitted on 21st MAI 2024)		  		
- **M.Sc.**,  Network and Distributed Systems | <a href="https://tnau.ac.in/" target="_blank">University of Amar Telidji of Laghouat</a>  			        		
- **B.Sc.**, Fundamental Computer Science | <a href="https://cattheni.edu.in/" target="_blank">University of Ziane Achour Djelfa</a>

## üî¨ Research Experience

<!---**Ph.D. research @<a href="https://en.ustb.edu.cn/" target="_blank">USTB</a> (_2018 - 2024_)**
- Developed and applied advanced tissue culture techniques like embryo culture to rapidly generate **somaclonal variants of rice** with improved salinity stress tolerance.
- Identified and validated novel genomic regions **(QTLs)** governing salinity tolerance through **molecular mapping** in rice F<sub>2</sub> populations along with thorough phenotypic and biochemical characterization.
- Employed **multi-omics approaches** including GC-MS metabolomics to uncover key genes and metabolites linked to superior osmotic/ionic adjustment under salt stress.
-->
**Ph.D. research @<a href="https://en.ustb.edu.cn/" target="_blank">USTB</a> (_2018 - 2024_)**
- Developed innovative solutions leveraging IoT, artificial intelligence, and computing offloading to tackle real-world challenges.
- Created a facial recognition application utilizing nearby idle devices for efficient task allocation, optimizing latency and communication costs.
- Deployed fog nodes in high-traffic urban areas to enhance public safety.
- Implemented UAV fog nodes in disaster-stricken areas to support evacuation and improve emergency communication infrastructure.
- Advanced the application of cutting-edge technologies through these projects, addressing critical challenges and driving innovation.

<!--- Single plant phenotying of diverse germplasm accessions (Sorghum, Peral millet, Pigeon pea ‚Äì genotyped 1,980 plants/samples combined) to examine intra and inter accession genetic diversity. 
- Single plant genotyping of accessions using **<a href="https://www.diversityarrays.com/" target="_blank">DArTSeq</a> based SNPs**.
- Harnessed whole genome genotyping, simulations, and predictive modeling to provide integrated insights into the dynamics of genetic variation across a species' range, by quantifying genomic diversity within and between isolated subpopulations, and developed optimized statistical frameworks to guide sustainable sampling regimes that limit genetic drift. -->

## üíº Work experience 
**Data Science Consultant @ <a href="https://www.fiverr.com/allanvictor_?public_mode=true" target="_blank">Fiverr</a> (_Decembe 2020 - Present_)**
- 5‚≠ê rated freelancer with over 200 hours of data analysis and visualization projects completed in R with over six years of experience specializing in genomic data analysis and bioinformatics.
- Proficient in handling big data and performing complex modelling using popular R libraries and packages such as tidyr, data.table, dplyr, plyr, tenserflow, ggplot, ggdendro, ggtree, ggheatmap, and circos. 
- Received positive feedback from clients for knowledge, professionalism, and mastery of R programming. Demonstrated ability to deliver high-quality work as evidenced by the portfolio available on the <a href="https://www.fiverr.com/allanvictor_?public_mode=true" target="_blank">Profile</a>. 

## üíª Programming and data analysis skills
- Proficient in full stack development of **R packages** using modular coding practices.
- Created **production-grade Shiny web applications** for interactive data analysis and visualization and; expertise in dependency management tools like Golem for scalable deployment.
- **Machine learning model** building for image classification/segmentation tasks; trained CNNs and other deep learning architectures in R, Python (PyCharm).
- **Multivariate data analysis** of large-scale omics datasets including genomics, phenomics and metabolomics using cutting-edge bioinformatics tools.
- Experience with analysis of **next-generation sequencing** data including quality control, read mapping, variant calling, expression quantification, metagenomic profiling, and associated statistical analysis using standard workflows in R and Python.
- **Advanced visualization** for multi-dimensional biological data through Circos, ggtree, ggtreeextra, Cytoscape and other platforms.

**Additional Skills:**
  
- Git/GitHub for version control and collaborative coding.
- High performance computing on clusters for scalable data analysis.
- Bioconductor for analyzing genomic/transcriptomic experiments.
- Workflow automation to enhance reproducibility and to faster workflow. 

## üå± R Packages developed
### ‚úÖ PBGeno
<a href="https://github.com/Allan-gitrepos/pbgeno" target="_blank">GitHub</a>

Developed pbgeno, an R package to streamline data analysis workflows for plant breeders. The package provides functions for calculating genetic distances, clustering genotypes, estimating diversity statistics, creating publication-quality visualizations, and automating routine tasks. Key features include calculating genetic distance matrices, structure-based clustering, polymorphism quantification, and converting proprietary marker genotypes into standardized formats for genome-wide association mapping. 

### ‚úÖ PBPerfect
**PBPerfect** <a href="https://allanbiotools.shinyapps.io/pbperfect/" target="_blank">(Visit Page)</a> is a interactive web tool enabling reproducible multivariate analysis with visualization of phenotypic and genotypic data. It features basic statistics, experimental designs, SSR workflows, multivariate analysis, mating designs, and dynamic graphics with outputs exported as publication-standard tables and graphics requiring no further formatting.

 Here is a draft GitHub readme section for your PBlinkagemap package:

### ‚úÖ PBlinkagemap

**PBlinkagemap** <a href="https://allanbiotools1.shinyapps.io/PBLinkagemap/" target="_blank">(Visit Page)</a> enables easy creation of linkage maps and identification of associated quantitative trait loci (QTLs) from genomic and phenotypic datasets. 

It allows users to:
- Import chromosome, marker, map distance and trait score data
- Interactively explore results on linkage maps
- Visualize QTL locations and effects
By handling computationally intensive linkage analysis and mapping behind the scenes, PBlinkagemap makes it simple for users to go from datasets to QTL discovery through an intuitive interface.

### ‚úÖ PB-GWAS üß¨

**PB-GWAS** <a href="https://allanbiotools1.shinyapps.io/PBgwas/" target="_blank">(Visit Page)</a> makes powerful genome-wide association studies accessible through an easy-to-use web app üë©‚Äçüíª

Key features:
üì• File upload in 4 clicks  
üèÉ‚Äç‚ôÇÔ∏è One-click GWAS launch  
‚öôÔ∏è Adjust parameters via sidebar   
üìà Interactive result plotting  
üìÑ Full PDF report downloading

By eliminating coding barriers, PB-GWAS allows both new and advanced users to leverage GAPIT workflows with no programming expertise required!. Whether you want to map simple or complex traits, PB-GWAS provides the automated analysis to accelerate discoveries üî¨

 Here is a draft GitHub readme for the PBHaploMineR package:

### ‚úÖ PBHaploMineR üß¨

**PBHaploMineR** <a href="https://github.com/Allan-gitrepos/PBhaplominerR" target="_blank">(Visit Page)</a> provides a toolkit to streamline pangenome haplotype mining and comparison from next-generation sequencing data. This R package aims to make large-scale haplotype analysis efficient and accessible for species with reference pangenomes.

**Key Features:**
* **Sequence Import** - Functions to import raw reads from multiple platforms and store in standardized schema
* **Haplotype Calling** - Optimized algorithms for pangenome-wide haplotype calling, incorporating structural variation
* **HapViz** - Interactive visualization system to explore and compare haplotypes in context of pangenome structure
* **HapCompare** - Statistically compare haplotypes between groups of samples/accessions and identify associated genomic signatures
* **Parallelization** - Built-in parallelization to scale analyses across HPC infrastructure

PBHaploMineR is still under development and testing. ETA for first stable version is Q1 2024.

## üé§ Workshop and Conferences



![image](https://github.com/Naouri17/profile/assets/40064898/5ce0817c-1ae9-44b3-98ca-31ad0cec051e)


- Attended IEEE International Intelligent Internet of Things Conference 2023 on <a href="https://www.ieee-smartiot.org/" target="_blank"> Internet of Things (IoT)"</a> at Xining, China.
  
- Presented paper titled "**Task Offloading for Smart Glasses in Healthcare: Enhancing Detection of Elevated Body Temperature**" at 1<sup>th</sup> SmartIoT 2024 conference, Xining, China.

- Presented paper titled "**The granger causality analysis between acupoints in the same meridian based on skin conductance**" at 1<sup>th</sup> SmartIoT 2024 conference , Xining, China.
  
- I attended and participated in the AngelHack Beijing Hackathon 2019. AngelHack welcomes all start-ups, developers, entrepreneurs, and designers to their annual event. The main goal of the hackathon is to create opportunities for innovation.

## ‚úçÔ∏è Articles & Blogs
- <a href="https://medium.com/@victorallan" target="_blank">Medium Articles</a>

## üìú Publications

I have published several publications in top-tier journals , focusing on areas such as fog computing, edge computing, IoT applications, federated learning, and service recommendation systems. Some of my notable publications include:

Naouri, Abdenacer; Wu, Hangxing; Nouri, Nabil Abdelkader; Dhelim, Sahraoui; Ning, Huansheng. **"A novel framework for mobile-edge computing by optimizing task offloading." IEEE Internet of Things Journal 8(16): 13065-13076 (2021). IEEE.** (DOI: <a href="https://doi.org/10.1109/JIOT.2021.3064225" target="_blank"> https://doi.org/10.1109/JIOT.2021.3064225</a>) | **Impact Factor: 10.6**
   
Naouri, Abdenacer; Ning, Huansheng; Nouri, Nabil Abdelkader; Khelloufi, Amar; Sada, Abdelkarim Ben; Naouri, Salim; Qammar, Attia; Dhelim, Sahraoui. **"Maximizing UAV Fog Deployment Efficiency for Critical Rescue Operations." Future Generation Computer Systems:1-16 (2024).** (DOI: <a href="https://doi.org/10.1016/j.future.2024.05.007" target="_blank"> https://doi.org/10.1016/j.future.2024.05.007</a>) |  **Impact Factor: 7.5** 

Naouri, Abdenacer, Nabil Abdelkader Nouri, Amar Khelloufi, Abdelkarim Ben Sada, Huansheng Ning, and Sahraoui Dhelim. **"Efficient fog node placement using nature-inspired metaheuristic for IoT applications." Cluster Computing (2024): 1-17.** (DOI: <a href="https://doi.org/10.1007/s10586-024-04409-3" target="_blank"> https://doi.org/10.1007/s10586-024-04409-3 </a>) |  **Impact Factor: 4.4**

Naouri, Abdenacer; Nouri, Nabil Abdelkader; Khelloufi, Amar; Sada, Abdelkarim Ben; Naouri, Salim; Ning, Huansheng; Dhelim, Sahraoui. **"BusCache: V2V-based infrastructure-free content dissemination system for Internet of Vehicles." IEEE Access (2024). IEEE.** (DOI: <a href="https://doi.org/10.1109/ACCESS.2024.3374881" target="_blank"> https://doi.org/10.1109/ACCESS.2024.3374881 </a>)  |  **Impact Factor: 4.0**

**And i have contributed on more than 20 research paper**.

## üìö References

|     |     |
|:----|:----|
| **Name:** | **Prof. Huansheng Ning** |
| **Position:** | Associate Dean of School of University of Science and Technology Beijing (USTB) |
| **Organization:** | University of Science and Technology Beijing (USTB), Beijing, China |
| **E-mail:** | <a href="mailto:azevedovcr@gmail.com">ninghuansheng@ustb.edu .cn</a> |
| **Home Page:** | <a href="http://www.cybermatics.org/" target="_blank">Visit Page</a> |

|     |     |
|:----|:----|
| **Name:** | **Dr. Sahraoui Dhelim** |
| **Position:** |  Scientist Researcher, IoT |
| **Organization:** | University College Dublin, Ireland |
| **E-mail:** | <a href="mailto:M.Vetriventhan@cgiar.org">sahraoui.dhelim@hotmail.com</a> |
| **Home Page:** | <a href="https://sahraoui17.github.io/profile/" target="_blank">Visit Page</a> |

|     |     |
|:----|:----|
| **Name:** | **Prof. Nabil Nouri** |
| **Position:** | Professor, Intelligent systems, Machine learning, Metaheuristic |
| **Organization:** | University of Djelfa, Algeria |
| **E-mail:** | <a href="mailto:wishmeguru@gmail.com">wishmeguru@gmail.com</a> |
| **LinkedIn:** | <a href="https://www.linkedin.com/in/rajaguru-bohar-5038762a/" target="_blank">Visit Page</a> |
|     |     |


## üìû Contact

| **Name**   | Abdenacer. N | 
| **E-mail**   | <a href="mailto:albertoogy@gmail.com">nacer.naouri@gmail.com</a> | 
| **LinkedIn**   | <a href="https://www.linkedin.com/in/allan-victor-6919b4152/" target="_blank">Visit Page</a> | 
